### PR TITLE
Add a Button to Allow Hide/Show of Sensitive Information

### LINKED ISSUE #5501: Allow unhiding of passwords/sensitive information
from the sig ui discussion: we need to collect all the places it’s relevant, search for locations we already have asterisks/hiding functionality, low priority

### PR DESCRIPTION
**Pull request by ahmadhamzh**
**What this PR does / why we need it**:
Add a Button to Allow Hide/Show input value in all places where we have Sensitive Information

image

**Which issue(s) this PR fixes**:
Fixes #5501 

**What type of PR is this?**
/kind design

```release-note
NONE
```

```documentation
NONE
```

### CODE REVIEW

**Comment by ahmedwaleedmalik** [comment]
@ahmadhamzh can you please add a gif/screenshot for how this looks like?

**Comment by Waseem826** [comment]
@ahmadhamzh wouldn't it be better to create a separate component for password fields instead of creating a directive and then adding mask button programmatically?

**Comment by Waseem826** [comment]
As far as custom hints and error messages are concerned, we can project them from outside using `ng-content` similar to how we do in `km-autocomplete` and `km-chip-list` components. And I think it would be better to create a component instead of directive because directives are generally used for changing some behaviour of host element but here we're always adding a new element as sibling of host element.

**Comment by ahmedwaleedmalik** [comment]
> directives are generally used for changing some behaviour of host element but here we're always adding a new element as sibling of host element

Which is something completely fine to do with directives. Basically, this use case, show/hide passwords, is mostly solved using directives. For me, I'm divided between using a directive or a dedicated component but I'd lean more towards using a directive.

**Comment by Waseem826** [comment]
@ahmadhamzh upon some rethinking, its fine to use directive approach.

File: modules/web/src/app/shared/directives/input-password.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  addButton() {
+    const input = this._el.nativeElement;
+    input.type = 'password';
+    const button = document.createElement('button');
+    button.classList.add('mat-icon-password-show-button', 'matSuffix');
+    const icon = document.createElement('i');
+    icon.classList.add('km-icon-mask', this._showPassword ? 'km-icon-show' : 'km-icon-hide');
+    button.appendChild(icon);
+    button.addEventListener('click', event => {
```
Comment (Waseem826) [inline modules/web/src/app/shared/directives/input-password.ts:37]: Please make sure to remove this event listener in `ngOnDestroy`

**Review by Waseem826 — APPROVED** [review]

**Pull request merged**